### ISSUE TITLE
Support redirecting to any page after "login"

### ISSUE THREAD

**Issue by chmanie** [JoinColony/colonyDapp#929]
Currently when a user directly visits a page (e.g. `/colony/COLONY_NAME`) they are redirected to `/connect`. But after they are connected they are always redirected to `/dashboard` and not to the page they requested originally.

To solve this we can use the router's `state` to store the page we want to redirect to. We already are doing that in the beta [here](https://github.com/JoinColony/colonyBeta/blob/012e3c221f17a500c161543a50395f4b478594ea/src/client/modules/users/components/Login/Login.js#L56) and [here](https://github.com/JoinColony/colonyBeta/blob/012e3c221f17a500c161543a50395f4b478594ea/src/client/modules/users/routeHooks.js#L11).

Basically before we redirect to `/connect` because the user is not logged in, we store `{ redirectTo: location.pathname + location.search }` in the state and when we redirect after logging in we read this value from the state and redirect the user there.
